Sources/StreamChatSwiftUI/ChatChannel/Composer/VoiceRecording/LockedView.swift

```diff
@@ -0,0 +1,123 @@
+                }
+                RecordingDurationView(duration: viewModel.audioRecordingInfo.duration)
+                RecordingWaveform(
+                    duration: viewModel.audioRecordingInfo.duration,
+                    currentTime: viewModel.recordingState == .stopped ?
+                        voiceRecordingHandler.context.currentTime :
+                        viewModel.audioRecordingInfo.duration,
+                    waveform: viewModel.audioRecordingInfo.waveform
+                )
+                Spacer()
+            }
+            .padding(.horizontal, 8)
+
+            HStack {
+                Button {
+                    withAnimation {
+                        viewModel.discardRecording()
+                    }
+                } label: {
+                    Image(systemName: "trash")
+                }
+
+                Spacer()
+                
+                if viewModel.recordingState == .locked {
+                    Button {
+                        withAnimation {
+                            viewModel.previewRecording()
+                        }
+                    } label: {
+                        Image(systemName: "stop.circle")
+                            .foregroundColor(.red)
+                    }
+                    
+                    Spacer()
+                }
+                
+                Button {
+                    withAnimation {
+                        viewModel.confirmRecording()
+                    }
+                } label: {
+                    Image(systemName: "checkmark.circle.fill")
+                }
+            }
+            .padding(.horizontal, 8)
+        }
+        .background(Color(colors.background).edgesIgnoringSafeArea(.bottom))
+        .offset(y: -20)
+        .background(Color(colors.background).edgesIgnoringSafeArea(.bottom))
+        .overlay(
+            viewModel.recordingState == .locked ? TopRightView {
+                LockedRecordIndicator()
+            }
+                : nil
+        )
+        .onAppear {
+            player.subscribe(voiceRecordingHandler)
+        }
+        .onReceive(voiceRecordingHandler.$context, perform: { value in
+            if value.state == .stopped || value.state == .paused {
+                isPlaying = false
+            } else if value.state == .playing {
+                isPlaying = true
+            }
+        })
+    }
+    
+    private func handlePlayTap() {
+        if isPlaying {
+            player.pause()
+        } else if let url = viewModel.pendingAudioRecording?.url {
+            player.loadAsset(from: url)
+        }
+        isPlaying.toggle()
+    }
+}
+
+struct LockedRecordIndicator: View {
+    @Injected(\.colors) var colors
+    
+    var body: some View {
+        Image(systemName: "lock")
+            .padding(.all, 8)
+            .background(Color(colors.background6))
+            .foregroundColor(.blue)
+            .clipShape(Circle())
+            .offset(y: -66)
```

Comment:
:( Is that necessary? I'm not a huge fan of offsets as they break the `expected` area/size of the view

Reply:
Not a fan either, but I don't see any other way. The composer and the message list can be used standalone (they're independent), and this is the only way to have the locked view showing in the message list space. 
Small price to pay, but the benefits of having them separated are large.